deployment/params.py

```diff
@@ -105,6 +101,18 @@ def _resolve_deployer() -> str:
         return deployer_account.address
 
 
+def _validate_transaction_args(args: typing.Tuple[Any, ...], abi) -> typing.Dict[str, Any]:
+    """Validates the transaction arguments against the function ABI."""
+    named_args = dict()
+    for arg, abi_input in zip(args, abi.inputs):
```

Comment:
Should we double-check that the length of `args` and `abi.inputs` are the same? Does `zip` fail if the lengths are different, or does it drop values?

Reply:
Fair point... unfortunately by default `zip` will silently carry on even if the inputs are different lengths...

There is a way to enable this check In python 3.10+: 

> [PEP 618](https://www.python.org/dev/peps/pep-0618): The [zip()](https://docs.python.org/3.10/library/functions.html#zip) function now has an optional strict flag, used to require that all the iterables have an equal length.

```suggestion
    for arg, abi_input in zip(args, abi.inputs, strict=True):
```

Reply:
👍 If we don't want to worry about python version, we could just check the `len(args) == len(abi.inputs)`. We do something similar in `_validate_constructor_abi_inputs(...)`.

Reply:
tbh it will be great to just support the latest python in this repo since we have the flexibility

Reply:
Yep, I'm fine either way - just giving options.

Reply:
While working in adding the suggested check I realized 3 things:
1. The check is redundant because the prior calling this function, the `_get_function_abi` called just before that gives you the abi that has exactly the required number of arguments.
2. The current approach is brittle as Solidity has function polymorphism, i.e., many functions can share the same name, as long as they have different parameter types (or number of parameters), and we're assuming that we can make a distinction based solely on number of parameters.
3. While I'm fixing 1 and 2 in an additional commit to this PR (fe592c3a37a174cc1837274a5bc5fab758e07017), I think we're very likely over-validating stuff that web3py and ape already do, and consequently, unnecessarily increasing code complexity. Not for this PR, but we probably should subtract a little.

Reply:
The underlying issue is that we try to do eager validation **before** executing a contract call (eager validation). Web3py is great for validation when actually calling a contract function, but not so much on the eager side. We do the eager validation on purpose because our scripts do multiple deployments/actions, and having it fail in the middle during the contract call is annoying, and potentially problematic, so we try the eager check to reduce such simplistic errors.

That being said, I don't doubt that we can better leverage web3py than we do currently, perhaps reuse common validation calls that web3 py uses to reduce our own logic - something like https://github.com/ethereum/web3.py/blob/main/web3/_utils/contracts.py#L144. Subtracting a little would be good.